src/app/components/room/room.component.ts

```diff
@@ -1,15 +1,99 @@
 import { Component, OnInit } from '@angular/core';
+import {readPost, User, writePost } from 'src/app/shared-module/models/user';
+import { LoginService } from 'src/app/services';
+import { Constants } from 'src/app/shared-module/constants/constants';
+import { Router } from '@angular/router';
+import { ToasterService } from 'src/app/core-module/services';
+import { isJSDocThisTag } from 'typescript';
+
 
 @Component({
   selector: 'app-room',
   templateUrl: './room.component.html',
   styleUrls: ['./room.component.scss']
 })
 export class RoomComponent implements OnInit {
+  messageInput: string = "";
+  posts: readPost[];
 
-  constructor() { }
+  constructor(private loginService: LoginService, private toaster: ToasterService, private router: Router) {
+    this.getPosts();
```

Comment:
In Angular you never want to call a method from the constructor, it should always live within ngOnInit. Could you move this line to that method?